sample_information_validator.py

```diff
@@ -51,44 +60,97 @@ def validate(self, cell):
     validates them individually.
     """
-
+    if(sheet[plate_id].value == None or sheet[plate_id].value == 'P####P#'):
+        logger.error(
+            'Missing PLATE ID'
+            )
     validator = NumericValidator(**validator_attr)
     passes = 0
     total = 0
-    for row_nr in range(row_start, row_end):
+    for row_nr in sample_rowID:
         total += 1
-        cell_id = "{col}{row_nr}".format(col=column_letter, row_nr=row_nr)
-        result = validator.validate(sheet[cell_id])
-        if result:  # Test passed
-            passes += 1
+        cell_id_conc = "{col}{row_nr}".format(col=concentration_letter, row_nr=row_nr)
+        cell_id_vol = "{col}{row_nr}".format(col=volume_letter, row_nr=row_nr)
+        cell_id_rin = "{col}{row_nr}".format(col=rin_letter, row_nr=row_nr)
+        result_conc = validator.validate_conc(sheet[cell_id_conc])
+        result_vol = validator.validate_vol(sheet[cell_id_vol])
+        if (sheet[cell_sample_type].value == 'Total RNA' or sheet[cell_sample_type].value == "mRNA" or sheet[cell_sample_type].value == 'Small RNA'):
+            result_rin = validator.validate_rin(sheet[cell_id_rin], sheet[cell_sample_type])
+            if result_conc and result_vol and result_rin:  # Test passed
+                passes += 1
+        else:
+            if result_conc and result_vol:  # Test passed
+                passes += 1
+    if (sheet[cell_sample_type].value == 'Total RNA' or sheet[cell_sample_type].value == "mRNA" or sheet[cell_sample_type].value == 'Small RNA'):
+        logger.info(
+            'Checked columns {}, {} and {}. {}/{} passes'.format(concentration_letter, volume_letter, rin_letter, passes, total)
+            )
+    else:
+        logger.info(
+            'Checked columns {} and {}. {}/{} passes'.format(concentration_letter, volume_letter, passes, total)
+            )
 
-    logger.info(
-        'Checked column {}. {}/{} passes'.format(column_letter, passes, total)
-        )
 
+def sample_number(sheet, sample_letter, row_start):
+    """ identifies the all rows containing a sample name"""
+    real = 1
+    cellID_withSample =list()
+    cellID_noSample =list()
+    for i in range(row_start, row_start+96):
+        cell_id = "{col}{row_itter}".format(col=sample_letter,row_itter=i)
+        if(sheet[cell_id].value != None):
+            cellID_withSample.append(i)
+        else:
+            cellID_noSample.append(cell_id)# TODO check here that these rows do really not contain information
+    return(cellID_withSample)
 
 def main(input_sheet):
     wb = load_workbook(input_sheet, read_only=True, data_only=True)
     ws = wb[SHEET_NAME]
-    nr_samples = 8  # TODO
-    final_row = FIRST_LINE + nr_samples
-    validate_column(ws, CONC_COL, FIRST_LINE, final_row, NumericValidator)
+    nr_sample_pos = sample_number(ws, SAMPLE_NAME_COL, FIRST_LINE)
+    validate_column(ws, PLATE_ID, CONC_COL, VOL_COL, RIN_COL, SAMPLE_TYPE, nr_sample_pos, NumericValidator)
```

Comment:
The all caps variables are global and you don't actually have to send them through to the function like this. But maybe its to be extra transparent a reader?

Reply:
Yes, its for readability. I can change it if you want?